### PR TITLE
Feature layout

### PR DESCRIPTION
**Pull request by gaimanjing**
### Description
Implement flex and grid layouts

### Explanation of Changes
Lay out first level node when window resizes

### CODE REVIEW

File: src/Domain/Layout/Node.cpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+{
+  // updateAt
+  auto viewModel = m_viewModel.lock();
+  if (!viewModel)
+  {
+    return;
+  }
+
+  nlohmann::json::json_pointer path{ m_path };
+  nlohmann::json objectJson{ viewModel->content()[path] };
```
Comment (elff1) [inline src/Domain/Layout/Node.cpp:179]: same as previous

File: src/Application/src/Controller.cpp [hunk 2]
```diff
@@ -285,30 +268,52 @@ JsonDocumentPtr Controller::wrapJsonDoc(std::shared_ptr<JsonDocument> jsonDoc)
...
 {
-  StartRunning start_running{ model };
-  start_running.layout(size);
+  std::string jsonSchemaFilePath;
+  if (pJsonSchemaFilePath)
+  {
+    jsonSchemaFilePath.append(pJsonSchemaFilePath);
+  }
+
+  auto lambda = [&, jsonSchemaFilePath](const json& designJson)
```
[the remark this answers is not in the document]
Comment (gaimanjing) [reply src/Application/src/Controller.cpp]: `jsonSchemaFilePath` is used in the returned lambda, so copy it.

File: src/Domain/Layout/Node.cpp [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    auto oldSize = m_frame.size;
+    auto newSize = frame.size;
+
+    m_frame = frame;
+
+    // updateAt
+    if (auto viewModel = m_viewModel.lock())
+    {
+      nlohmann::json::json_pointer path{ m_path };
+      nlohmann::json objectJson{ viewModel->content()[path] };
```
[the remark this answers is not in the document]
Comment (gaimanjing) [reply src/Domain/Layout/Node.cpp]: fixed

File: src/Domain/Layout/Node.cpp (hunk 1 above)
Comment (gaimanjing) [reply src/Domain/Layout/Node.cpp]: fixed

**Pull request merged**